javascript/ql/src/Declarations/MixedStaticInstanceThisAccess.ql

```diff
@@ -10,15 +10,20 @@
 */
 import javascript
 
+MethodDefinition getASuperClassMethod(ClassDefinition base, string name) {
+  result.getDeclaringClass() = base and
+  result.getName() = name
+  or
+  result = getASuperClassMethod(base.getSuperClass().analyze().getAValue().(AbstractClass).getClass(), name)
```

Comment:
Can we pull out the `base.getSuperClass()...` bit into a member predicate of `ClassDefinition`?